src/signals/incident-management/components/CalendarInput/__tests__/CalendarInput.test.js

```diff
@@ -1,73 +1,77 @@
+  const calendarInputProps = {
+    id: 'foo',
+    label: 'Here be dragons',
+    name: 'my_date_field',
+    onSelect: () => {},
+  };
+  const id = 'bar';
+
   it('renders a datepicker component', () => {
     render(withAppContext(<CalendarInput {...calendarInputProps} />));
 
-    expect(
-      document.querySelectorAll('[class*=react-datepicker]').length
-    ).toBeGreaterThan(0);
+    expect(document.querySelectorAll('[class*=react-datepicker]').length).toBeGreaterThan(0);
   });
 
   it('renders a CustomInput component', () => {
-    const { getByTestId } = render(
-      withAppContext(<CalendarInput {...calendarInputProps} />)
-    );
+    render(withAppContext(<CalendarInput {...calendarInputProps} />));
 
-    expect(getByTestId('calendarCustomInputElement')).toBeInTheDocument();
+    expect(screen.getByTestId('calendarCustomInputElement')).toBeInTheDocument();
   });
 
-  it('renders a hidden input field', () => {
-    const { rerender } = render(
-      withAppContext(<CalendarInput {...calendarInputProps} />)
-    );
+  it('renders the selected date in the input field', () => {
+    const { rerender } = render(withAppContext(<CalendarInput {...calendarInputProps} />));
 
-    expect(
-      document.querySelector('input[type=hidden]')
-    ).not.toBeInTheDocument();
+    const element = screen.getByTestId('calendarCustomInputElement');
+    expect(element.querySelector('input').value).toEqual('');
 
     const selectedDate = new Date();
 
-    rerender(
-      withAppContext(
-        <CalendarInput {...calendarInputProps} selectedDate={selectedDate} />
-      )
-    );
+    rerender(withAppContext(<CalendarInput {...calendarInputProps} selectedDate={selectedDate} />));
 
-    expect(document.querySelector('input[type=hidden]')).toBeInTheDocument();
+    const elementWithDate = screen.getByTestId('calendarCustomInputElement');
+    expect(element.querySelector('input').value).toEqual(dateToString(selectedDate));
   });
 
   it('should call onSelect', () => {
     const onSelect = jest.fn();
-    const id = 'qux';
 
-    render(
-      withAppContext(
-        <CalendarInput {...calendarInputProps} id={id} onSelect={onSelect} />
-      )
-    );
+    render(withAppContext(<CalendarInput {...calendarInputProps} id={id} onSelect={onSelect} />));
 
     const inputElement = document.getElementById(id);
 
     expect(onSelect).not.toHaveBeenCalled();
 
     fireEvent.change(inputElement, { target: { value: '18-12-2018' } });
 
-    expect(onSelect).toHaveBeenCalledWith(
-      expect.any(Date),
-      expect.any(Object)
-    );
+    expect(onSelect).toHaveBeenCalledWith(expect.any(Date));
+  });
+
+  it('should focus on the input when a value is selected', () => {
+    const onSelect = jest.fn();
+    const selectedDate = new Date();
+
+    render(withAppContext(<CalendarInput {...calendarInputProps} id={id} onSelect={onSelect} />));
+
+    const inputElement = document.getElementById(id);
```

Comment:
You can use `screen` here: 

`const inputElement = screen.getByRole('textbox', { name: calendarInputProps.name });`

Reply:
Yeah, try to avoid `*ByTestId` and `querySelector`.

Reply:
I have changed it but it doesn't fill right. When selecting byRole, the `name` option refers to the aria accessibility name which in this case is the `label` prop 👎🏻